addons/web/static/tests/views/fields/integer_field_tests.js

```diff
@@ -286,4 +287,31 @@ QUnit.module("Fields", (hooks) => {
             "Placeholder"
         );
     });
+
+    QUnit.test(
+        "a field is considered not dirty when the change event is triggered",
+        async function (assert) {
+            // This test is introduced to fix a bug:
+            // Have a valid value, change it to an invalid value, blur, then change it back to the same valid value.
+            // The field was considered not dirty, so the onChange code wasn't executed, and the model still thought the value was invalid.
+            await makeView({
+                type: "form",
+                serverData,
+                resModel: "partner",
+                resId: 1,
+                arch: '<form><field name="int_field"/></form>',
+            });
+
+            const input = findElement(target, ".o_field_widget[name=int_field] input");
```

Comment:
I'm always careful with storing nodes in variables and interacting with them later on in the test. We can't guarantee that the node won't be replaced when the view is re-rendered, so I think it's safer to always querySelect.

Also, I'm not sure that findElement is necessary here. I would simply use querySelector